ctl/waypoint/waypoint.go

```diff
@@ -0,0 +1,628 @@
+	defer ticker.Stop()
+	if namespace == "" {
+		fmt.Fprintln(w, "NAMESPACE\tNAME\tSTATUS\tTYPE\tREASON\tMESSAGE")
+	} else {
+		fmt.Fprintln(w, "NAME\tSTATUS\tTYPE\tREASON\tMESSAGE")
+	}
+	for _, gw := range gw {
+		for range ticker.C {
+			programmed := false
+			gwc, err := kubeClient.GatewayAPI().GatewayV1().Gateways(namespaceOrDefault(namespace)).Get(context.TODO(), gw.Name, metav1.GetOptions{})
+			if err == nil {
+				// Check if gateway has Programmed condition set to true
+				for _, cond = range gwc.Status.Conditions {
+					if cond.Type == string(gateway.GatewayConditionProgrammed) && string(cond.Status) == "True" {
+						programmed = true
+						break
+					}
+				}
+			}
+			if namespace == "" {
+				fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\n", gwc.Namespace, gwc.Name, cond.Status, cond.Type, cond.Reason, cond.Message)
+			} else {
+				fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\n", gwc.Name, cond.Status, cond.Type, cond.Reason, cond.Message)
+			}
+
+			if programmed {
+				break
+			}
+			if time.Since(startTime) > waitTimeout {
+				return errorWithMessage("timed out while retrieving status for waypoint", gwc, err)
+			}
+		}
+	}
+	return w.Flush()
+}
+
+func labelNamespaceWithWaypoint(kubeClient kube.CLIClient, ns string) error {
+	nsObj, err := getNamespace(kubeClient, ns)
+	if err != nil {
+		return err
+	}
+	if nsObj.Labels == nil {
+		nsObj.Labels = map[string]string{}
+	}
+	nsObj.Labels[KmeshUseWaypointLabel] = waypointName
+	if _, err := kubeClient.Kube().CoreV1().Namespaces().Update(context.Background(), nsObj, metav1.UpdateOptions{}); err != nil {
+		return fmt.Errorf("failed to update namespace %s: %v", ns, err)
+	}
+	return nil
+}
+
+func getNamespace(kubeClient kube.CLIClient, ns string) (*corev1.Namespace, error) {
+	nsObj, err := kubeClient.Kube().CoreV1().Namespaces().Get(context.Background(), ns, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		return nil, fmt.Errorf("namespace: %s not found", ns)
+	} else if err != nil {
+		return nil, fmt.Errorf("failed to get namespace %s: %v", ns, err)
+	}
+	return nsObj, nil
+}
+
+func namespaceHasLabel(kubeClient kube.CLIClient, ns string, label string) (bool, error) {
+	nsObj, err := getNamespace(kubeClient, ns)
+	if err != nil {
+		return false, err
+	}
+	if nsObj.Labels == nil {
+		return false, nil
+	}
+	return nsObj.Labels[label] != "", nil
+}
+
+func namespaceHasLabelWithValue(kubeClient kube.CLIClient, ns string, label, labelValue string) (bool, error) {
+	nsObj, err := getNamespace(kubeClient, ns)
+	if err != nil {
+		return false, err
+	}
+	if nsObj.Labels == nil {
+		return false, nil
+	}
+	return nsObj.Labels[label] == labelValue, nil
+}
+
+func namespaceOrDefault(namespace string) string {
+	if namespace == "" {
+		return "default"
+	}
+
+	return namespace
+}
+
+func getKmeshWaypointImage() string {
+	if image != "" {
+		return image
+	}
+
+	ver := version.Get().GitVersion
+
+	return fmt.Sprintf("ghcr.io/kmesh-net/waypoint:v%s", ver)
```

Comment:
`ghcr.io/kmesh-net/waypoint` can be customized

Reply:
You could use `--image ` flag to customize waypoint image.